src/main/java/org/openrewrite/maven/AbstractRewriteBaseRunMojo.java

```diff
@@ -62,6 +62,29 @@ public abstract class AbstractRewriteBaseRunMojo extends AbstractRewriteMojo {
     @Nullable
     protected LinkedHashSet<String> options;
 
+    /**
+     * The level used to log changes performed by recipes.
+     */
+    @Parameter(property = "rewrite.recipeChangeLogLevel", defaultValue = "WARN")
+    protected LogLevel recipeChangeLogLevel;
+
+    protected void log(CharSequence content) {
```

Comment:
All fine with this change. Just wanted to explain why I implemented it the way I originally did.

Now you essentially have a method in an abstract class, available in all the different implementations, and it only communicates "log" and a string to be logged. It is very implicit that the actual log level is affected by a LogLevel field with very specific semantics. I didn't want to introduce a method named only "log", which is intended for a very specific use: to log something that changed due to a rewrite.

To me, it seems better to have a method which accepts an explicit log level (e.g. the configured recipeChangeLogLevel), because the knowledge about what is logged is at the site where one invokes the logging-method. Now one really has to _know_ that the method named only "log" is specifically for logging changes made from a recipe, instead of the method accepting the loglevel as a parameter and carrying out as instructed, and the knowledge about why a certain loglevel is used is kept at the call site.

This can be implemented as a method accepting the LogLevel in the abstract class, absolutely, and that would also align with my original "design". Externalizing it to a separate (package scoped) class is just my preference to not tie too much concerns into abstract classes and making them available through inheritance.

Just wanted to mention it for you to consider :)

Reply:
A (better IMHO) alternative would be to perhaps have
```java
protected void log(LogLevel level, CharSequence content) {
```
instead, as it would not tie the log-method implicitly to the `recipeChangeLogLevel`. And you avoid the separate class.

It would be very easy for other code to evolve to invoke the new log-method accepting only the message, and not logging anything related to "changes from rewriting", but gets the log level unintentionally configured from `recipeChangeLogLevel`.

Reply:
If you prefer not repeating (passing) `recipeChangeLogLevel` from the different logging sites, I would rename the logging-method specifically for its intended use, perhaps `logRecipeChange(CharSequence)` or something, to communicate its intended use. It is not a _general_ log method.

Reply:
Thanks for that context! I've adopted your proposal of `log(LogLevel, CharSequence)`